src/main/java/seedu/duke/operationlist/FlightList.java

```diff
@@ -3,64 +3,70 @@
 import seedu.duke.exceptions.SkyControlException;
 
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
 public class FlightList extends OperationList {
     public static int flightIndex = 0;
-
     private static final String FLIGHT_ADD_COMMAND = "flight add";
+    private static final String FLIGHT_ADD_DELIMITER = "flight add ";
     private static final String FLIGHT_DELETE_COMMAND = "flight delete";
-    private static final String FLIGHT_NUMBER_DELIMITER = "fn/";
-    private static final String AIRLINE_DELIMITER = "a/";
-    private static final String DESTINATION_DELIMITER = "d/";
-    private static final String DEPARTURE_TIME_DELIMITER = "t/";
-    private static final String GATE_NUMBER_DELIMITER = "gn/";
-    private static final String TERMINAL_DELIMITER = "tl/";
-    private static final String CHECK_IN_ROW_DELIMITER = "c/";
-    private static final String END_OF_INPUT = "empty";
+    private static final String FLIGHT_NUMBER_DELIMITER = " fn/";
+    private static final String AIRLINE_DELIMITER = " a/";
+    private static final String DESTINATION_DELIMITER = " d/";
+    private static final String DEPARTURE_TIME_DELIMITER = " t/";
+    private static final String GATE_NUMBER_DELIMITER = " gn/";
+    private static final String TERMINAL_DELIMITER = " tl/";
+    private static final String CHECK_IN_ROW_DELIMITER = " c/";
+
+    protected static final String EMPTY_STRING = "";
+    protected static final int AIRLINE_LENGTH_LIMIT = 22;
+    protected static final int DESTINATION_LENGTH_LIMIT = 22;
+    protected static boolean isFlightNumberPresent = false;
+    protected static boolean isDepartureTimePresent = false;
+    protected static boolean isGateNumberPresent = false;
+    protected String flightNumber;
+    protected String airline;
+    protected String destination;
+    protected String departureTime;
+    protected String gateNumber;
+    protected String terminal;
+    protected String checkIn;
 
     protected static int numOfFlights = 0;
 
     public static void checkCommandLength(String description) throws SkyControlException {
         if (description.isEmpty()) {
-            throw new SkyControlException(description);
+            throw new SkyControlException(ui.showEmptyDescriptionMessage());
         }
     }
 
-    public static String extractDetail(String command, String start, String end) {
+    public static String extractDetail(String command, String start, String end) throws SkyControlException {
+
         String extractedDetail;
-        if (end.equals("empty")) {
-            extractedDetail = command.substring(command.indexOf(start) + start.length());
-        } else {
-            extractedDetail = command.substring(command.indexOf(start) + start.length(), command.indexOf(end) - 1);
+        int startIndex = command.indexOf(start) + start.length();
+        int endIndex = command.lastIndexOf(end);
+        if (endIndex <= startIndex || endIndex < 0 || startIndex < 0) {
```

Comment:
Can replace the zeros with a constant variable to avoid magic numbers. Good use of Boolean logic. However, it would be good if you would encompass this logic into a single Boolean Logic. I.e. isEmptyDetail.